File: Benchmarking/stats/bammitest/euler14.java
public class euler14
{
	public static void main(String[] args)
	{
		int limit = Integer.parseInt(args[0]);
		seq(limit);
	}
	
	private static long seq(int n)
	{
		long max = 0;
		long maxNumber = 0;
		long countSeq;
		long temp;
		long current;
		for(int i = 1; i < n+1; i++)
		{
			countSeq = 1;
			temp = i;
			current = i;
			while(current != 1)
			{
				if(current % 2 == 0)
				{
					current = current / 2;
					countSeq += 1;
				}
				else
				{
					current = 3 * current + 1;
					countSeq += 1;
				}				
			}
			if(countSeq > max)
			{
				max = countSeq;
				maxNumber = temp;
			}
		}
		return maxNumber;
	}
}

File: Benchmarking/stats/bammitest/euler14.cpp
//
//  main.cpp
//  Euler14
//
//  Created by Sonni on 20/05/16.
//  Copyright © 2016 Sonni Jensen. All rights reserved.
//

#include <iostream>

void seq(long long n)
{
    long long max = 0;
    long long max_number = 0;
    for (unsigned int i = 1; i <= n; i++)
    {
        unsigned int count_seq = 1;
        unsigned int temp = i;
        unsigned int tmpI = i;
        while (tmpI != 1)
        {
            
            if (tmpI % 2 == 0)
                tmpI /= 2;
            else
                tmpI = 3 * tmpI + 1;
            
            count_seq++;
        }
        if (count_seq > max)
        {
            max = count_seq;
            max_number = temp;
        }
    }
    std::cout << max << std::endl;
    std::cout << max_number << std::endl;
}

int main(int argc, const char * argv[]) {
    seq(atoi(argv[1]));
    return 0;
}
